tests/GatewayTest.php

```diff
@@ -57,43 +59,55 @@ public function testPurchaseUsingStringAmount()
             'singleUseTokenId' => 'EFG789',
         ]);
 
-        $this->assertInstanceOf('Omnipay\SchoolEasyPay\Message\PurchaseRequest', $request);
+        $card = new CreditCard([
+            'firstName' => 'Bobby',
+            'lastName' => 'Tables',
+            'number' => '4444333322221111',
+            'cvv' => '123',
+            'expiryMonth' => '12',
+            'expiryYear' => '2017',
+            'email' => 'testcard@gmail.com',
+        ]);
+
+        $request->setCard($card);
+
+        $this->assertInstanceOf(PurchaseRequest::class, $request);
         $this->assertSame('10.00', $request->getAmount());
 
+        $request->setCardProxy('bdd4c12345e54b00b4e1a2b309442a07');
+        $request->setCustomerReference('ABC123');
+
         $data = $request->getData();
 
-        $this->assertEquals('payment', $data['transactionType']);
-        $this->assertEquals('10.00',   $data['principalAmount']);
-        $this->assertEquals('aud',     $data['currency']);
-        $this->assertEquals('ABC123',  $data['customerNumber']);
-        $this->assertEquals('456',     $data['orderNumber']);
-        $this->assertEquals('EFG789',  $data['singleUseTokenId']);
+        $this->assertEquals('bdd4c12345e54b00b4e1a2b309442a07', $data['cardProxy']);
+        $this->assertEquals('ABC123', $data['customerReference']);
+        $this->assertEquals('Bobby Tables', $data['customerName']);
+        $this->assertEquals('10.00', $data['paymentAmount']);
     }
 
-    public function testPurchaseUsingMoney()
+    public function testPurchaseUsingMoney(): void
     {
+        $card = new CreditCard($this->getValidCard());
         $request = $this->gateway->purchase([
             'currency' => 'AUD',
-            'customerNumber' => 'ABC123',
```

Comment:
Are these values meant to have changed in the test, is it still testing the same thing?

Reply:
The tests previously never worked.  They were duped from the westpac package but I presume never run.  These updates reflect the values that are provided / returned by school-easy-pay.